README.md

```diff
@@ -1,57 +1,157 @@
+*Creates only Spark Cluster for a given VNet and Storage account if the LLAP cluster is already created and vice versa.
+* Supports  WASB and ADLS_GEN2 storage types, However the stroage account should be created before running this tool i.e this tool does not create a new storage account if it does not exist.
+* Supports creation of secure HWC Clusters i.e clusters with Enterprise Security Pack.
+* Custom VNet is configured with minimal Inbound Rules, however if the VNet is already present then it can reused.
+* This Tool expects Active Directory Credentials for creating the HDI cluster.
 
-* Feature 1
-* Feature 2
-* ...
 
 ## Getting Started
 
 ### Prerequisites
 
-(ideally very short, if any)
-
-- OS
-- Library version
-- ...
-
-### Installation
-
-(ideally very short)
-
-- npm install [package name]
-- mvn install
-- ...
-
-### Quickstart
-(Add steps to get up and running quickly)
-
-1. git clone [repository clone url]
-2. cd [respository name]
-3. ...
-
-
-## Demo
-
-A demo app is included to show how to use the project.
-
-To run the demo, follow these steps:
-
-(Add steps to start up the demo)
-
-1.
-2.
-3.
-
-## Resources
-
-(Any additional resources or related projects)
-
-- Link to supporting information
-- Link to similar sample
-- ...
+- An active Azure Account for creating HDI Clusters.
+- [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest) should be installed.
+    - Run the below command for creating a new Service Principal or use the one which is already created.
+    ```bash 
+    az account set -s YOUR_SUBSCRIPTION
+    az ad sp create-for-rbac --name YOUR_SERVICE_PRINCIPAL_NAME --sdk-auth
+    ```
+    - Store the json info returned from the above command into to a file so that the clientID, tenenatID and clientSecret info can be used while configuring the tool.
+
+### Configuring the Tool
+
+We use a YAML file to configure this Tool in order to create the Spark and LLAP HDI Cluster's as part of Hive Warehouse Connector in the same VNet.
+
+ - The below YAML config file should be passed to this tool inorder to initiate the HWC cluster creation.
+ - The below configs shows all the configuration available for running this tool. You can also refer the conf folder inside this repo which has templates for standard and secure cluster configs.
+ 
+```yml
+type: SPARK_AND_LLAP # Optional, values can be either SPARK_ONLY or LLAP_ONLY. Default is SPARK_AND_LLAP
+clusterNamePrefix: Foo # This prefix is used while creating the cluster name. Only first three chars are used as prefix from this string.
+resourceGroup: foobar-rg # Resource group where the cluster needs to be created
+region: eastus2 # Region name where the cluster needs to be created, should be in smallcase without space 
+headNodeVMSize: STANDARD_D13_V2 # Any Standrad VM Size supported for Head Nodes
+workerNodeVMSize: STANDARD_D13_V2 # Any Standrad VM Size supported for Worker Nodes
+workerNodeSize: 3 # Size of worker nodes
```

Comment:
Currently, the tool uses the same VM Configuration for both Spark and LLAP Clusters. The recommendation is shared on this [doc](https://docs.microsoft.com/en-us/azure/hdinsight/hdinsight-supported-node-configuration). To keep it simple  we have the same VM size and worker nodes being used for both Spark and LLAP. Although customers can change the worker node size manually from portal later, they cannot change the SKU once configured.

Reply:
I think, it make sense to support different SKU type for LLAP and Spark especially since the recommended default SKU type is D13v2 for Spark and D14v2 for LLAP. Can someone else take it up as follow-up item.

Reply:
I have created this [issue ](https://github.com/Azure-Samples/HWC-Cluster-Creator-Tool/issues/2) as a follow up item to support different VM Types and Worker Nodes Sizes while creating Spark and LLAP Cluster.